include/clue/shared_mutex.hpp

```diff
@@ -356,24 +358,26 @@ class shared_lock {
 
     void lock() {
         mut_->lock_shared();
+        owns_ = true;
     }
 
-    void try_lock() {
-        mut_->try_lock_shared();
+    bool try_lock() {
+        return owns_ = mut_->try_lock_shared();
     }
 
     template<class Rep, class Period>
     bool try_lock_for(const ::std::chrono::duration<Rep,Period>& duration) {
-        return mut_->try_lock_shared_for(duration);
+        return owns_ = mut_->try_lock_shared_for(duration);
```

Comment:
For code style purpose, don't write ``return a = b``, instead please rewrite this as:
```c++
owns_ = mut_->try_lock_shared_for(duration);
return owns_;
```